rir/src/compiler/pir/type.h

```diff
@@ -345,6 +346,36 @@ struct PirType {
         return t_.r.includes(o.t_.r);
     }
 
+    void addAssumptions(Assumptions& given, int idx) const {
+        if (!maybeObj())
+            given.setNotObj(idx);
+        if (given.isEager(idx) && given.isNotObj(idx) && isScalar()) {
```

Comment:
I don't understand this. why are some properties read from `given`, some other properties from type, then some other properties are updated. I see no logic. If this is the particular ad-hoc thing, that the builder needs, then better just keep this code at the builder site. because putting it on type suggests, that this is some generic functionality, when I really don't understand what the reusable property of this function would be...